src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -83,16 +86,27 @@ public void initialize()
     {
         if (DatabaseDescriptor.getAuditLoggingOptions().enabled)
             registerAsListener();
+
+        if (!MBeanWrapper.instance.isRegistered(MBEAN_NAME))
+            MBeanWrapper.instance.registerMBean(this, MBEAN_NAME);
     }
 
-    private IAuditLogger getAuditLogger(ParameterizedClass logger) throws ConfigurationException
+    private IAuditLogger getAuditLogger(AuditLogOptions options) throws ConfigurationException
     {
-        if (logger.class_name != null)
+        final ParameterizedClass logger = options.logger;
+
+        if (logger != null && logger.class_name != null)
         {
             return FBUtilities.newAuditLogger(logger.class_name, logger.parameters == null ? Collections.emptyMap() : logger.parameters);
         }
 
-        return FBUtilities.newAuditLogger(BinAuditLogger.class.getName(), Collections.emptyMap());
+        return new BinAuditLogger(options);
+    }
+
+    @Override
+    public CompositeData getAuditLogOptionsData()
+    {
+        return AuditLogOptionsCompositeData.toCompositeData(AuditLogManager.instance.getAuditLogOptions());
```

Comment:
nit: And I leave it really to you to decide but I would probably put this method before or after` getAuditLogOptions`
The methods in between are short but this might change and I feel those two will be probably better together. Feel free to ignore me if you don't like the suggestion :-)